bitbots_vision/bitbots_vision/vision_modules/yoeo_handler.py

```diff
@@ -0,0 +1,207 @@
+from candidate import CandidateFinder, Candidate
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+import cv2
+from numpy import where, moveaxis
+from os.path import join
+from rclpy import logging
+from typing import List
+
+logger = logging.get_logger('vision_yoeo')
+
+try:
+    from yoeo import models as torch_models, detect as torch_detect
+    from yoeo.utils.utils import load_classes
+except ImportError:
+    logger.error("Not able to import pytorchyolo. This might be fine if you use another method.")
+
+
+class YOEOHandlerTemplate(ABC):
+    def __init__(self, config: dict, model_path: str):
+        self._use_caching = config['caching']
+
+        self._image = None
+        self._iou_non_max_suppression_thresh = config['yoeo_nms_threshold']
+
+        self._det_candidates = defaultdict(list)
+        self._det_class_names = None
+        self._det_confidence_thresh = config['yoeo_conf_threshold']
```

Comment:
Yeah, I was unsure with how the config would look like for YOEO. But since there is a nms_threshold for YOEO, it probably makes sense to have a common prefix, don't you think?

Reply:
I agree with you.